lib/drip/client/configuration.rb

```diff
@@ -0,0 +1,28 @@
+module Drip
+  class Client
+    class Configuration
+      DEFAULT_URL_PREFIX = "https://api.getdrip.com/".freeze
+      private_constant :DEFAULT_URL_PREFIX
+
+      CONFIGURATION_FIELDS = %i[access_token api_key account_id url_prefix http_open_timeout http_timeout].freeze
+
+      attr_accessor(*CONFIGURATION_FIELDS)
+
+      def initialize(**options)
+        remainder = options.keys - CONFIGURATION_FIELDS
+        raise ArgumentError, "unknown keyword#{'s' if remainder.size > 1}: #{remainder.join(', ')}" unless remainder.empty?
```

Comment:
This line is too heavy on my taste I would suggest to extract a method, e.g: 

```ruby
def initialize(**options)
  check_options!(options)

  ...
end

def check_options!(options)
  unknown_options = options.keys - CONFIGURATION_FIELDS
  return if unknown_options.empty?

  error_message = "unknown option"
  error_message << "s" if unknown_options.size > 1
  error_message << ": "
  error_message << unknown_options.join(', ')

  raise ArgumentError, error_message
end
```

Reply:
At this point, I wonder whether it is more complicated than it is worth, especially compared to the original parameter string, which did all of this automatically…

Reply:
What about this option?

```ruby
def initialize(**options)
  options.each do |k, v|
    public_send("#{k}=", v)
  end

rescue NoMethodError => e
  raise ArgumentError, "unknown option: #{e.name}"
end
```

Reply:
That could work too. I was originally trying to mimic how Ruby handles this by default, and try to avoid depending on exceptions to detect cases when not required.